contracts/loan_manager/src/contract.rs

```diff
@@ -57,8 +58,11 @@ impl LoanManager {
             .with_current_contract(salt)
             .deploy_v2(wasm_hash, ());
 
+        let admin: Address = e.storage().persistent().get(&LoansDataKey::Admin).unwrap();
+        admin.require_auth();
+
```

Comment:
Maybe that deploy pool test fails because the test doesn't init the manager like it should? This could probably be fixed by `e.mock_all_auths()` in that deploy_pool test.